File: 프로그래머스/lv2/42885. 구명보트/구명보트.py
from collections import deque
def solution(people, limit):
    people.sort()
    people = deque(people)
    result = 0
    while len(people) > 1:
        if people[0] + people[-1] <= limit:
            people.popleft()
        people.pop()
        result += 1
    return result + len(people)

File: 프로그래머스/lv2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    let answer = 0;
    people.sort((a,b) => a-b);
    while (people.length > 1) {
        if (people.pop() + people[0] <= limit) {
            people.shift();
        }
        answer++;
    }
    return answer + people.length;
}